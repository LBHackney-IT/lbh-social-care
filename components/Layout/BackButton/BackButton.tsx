import Router from 'next/router';

const BackButton = (): React.ReactElement => (
  <div className="lbh-container">
    <a
      className="govuk-back-link lbh-back-link lbh-link--no-visited-state"
      href="#"
      onClick={() => Router.back()}
    >
      Go back
    </a>
  </div>
);

export default BackButton;
