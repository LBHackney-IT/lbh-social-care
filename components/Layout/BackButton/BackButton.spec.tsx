import { render, fireEvent } from '@testing-library/react';

import Router from 'next/router';

import BackButton from './BackButton';

jest.mock('next/router', () => ({
  back: jest.fn(),
}));

describe('BackButton', () => {
  it('should render properly', async () => {
    const { asFragment } = render(<BackButton />);
    expect(asFragment()).toMatchSnapshot();
  });

  it('should trigger history back on click', () => {
    const { getByText } = render(<BackButton />);
    const button = getByText('Go back');
    fireEvent.click(button);
    expect(Router.back).toHaveBeenCalled();
  });
});
