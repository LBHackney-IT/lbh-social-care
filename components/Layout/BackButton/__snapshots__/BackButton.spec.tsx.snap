// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`BackButton should render properly 1`] = `
<DocumentFragment>
  <div
    class="lbh-container"
  >
    <a
      class="govuk-back-link lbh-back-link lbh-link--no-visited-state"
      href="#"
    >
      Go back
    </a>
  </div>
</DocumentFragment>
`;
